From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jagan Teki <jagan@amarulasolutions.com>
Date: Wed, 16 Feb 2022 16:03:15 +0530
Subject: [PATCH] arm64: dts: rockchip: Move USB Host enable on ROCK Pi 4 dts

This patch is trying to move USB host enable pins from common ROCK Pi
dtsi in to respective dts files in order to avoid conflict while
enabling Display Port.

Display Port on ROCK Pi 4C uses same RK_PD1 pin like other ROCK Pi 4
board used for USB host enable.

Enabling Display Port will conflict the pin usage as the RK_PD1 is
added globally in dtsi file.

Move it.

Signed-off-by: Jagan Teki <jagan@amarulasolutions.com>
Signed-off-by: Radxa <dev@radxa.com>
---
 .../arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi | 10 ----------
 .../arm64/boot/dts/rockchip/rk3399-rock-pi-4a.dts | 15 +++++++++++++++
 .../arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts | 15 +++++++++++++++
 .../arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts | 15 +++++++++++----
 4 files changed, 41 insertions(+), 14 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi
index d003cedb8b88..da0040db9f23 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi
@@ -84,10 +84,6 @@ vcc3v3_sys: vcc3v3-sys {
 
 	vcc5v0_host: vcc5v0-host-regulator {
 		compatible = "regulator-fixed";
-		enable-active-high;
-		gpio = <&gpio4 RK_PD1 GPIO_ACTIVE_HIGH>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&vcc5v0_host_en>;
 		regulator-name = "vcc5v0_host";
 		regulator-always-on;
 		vin-supply = <&vcc5v0_sys>;
@@ -573,12 +569,6 @@ vcc5v0_typec_en: vcc5v0-typec-en {
 		};
 	};
 
-	usb2 {
-		vcc5v0_host_en: vcc5v0-host-en {
-			rockchip,pins = <4 RK_PD1 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-	};
-
 	wifi {
 		wifi_enable_h: wifi-enable-h {
 			rockchip,pins = <0 RK_PB2 RK_FUNC_GPIO &pcfg_pull_none>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4a.dts b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4a.dts
index 89f2af5e111d..32f869c2e66b 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4a.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4a.dts
@@ -11,3 +11,18 @@ / {
 	model = "Radxa ROCK Pi 4A";
 	compatible = "radxa,rockpi4a", "radxa,rockpi4", "rockchip,rk3399";
 };
+
+&pinctrl {
+	usb2 {
+		vcc5v0_host_en: vcc5v0-host-en {
+			rockchip,pins = <4 RK_PD1 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+};
+
+&vcc5v0_host {
+	enable-active-high;
+	gpio = <&gpio4 RK_PD1 GPIO_ACTIVE_HIGH>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&vcc5v0_host_en>;
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts
index 1d4d73945012..4f16d11eb22a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts
@@ -12,6 +12,14 @@ / {
 	compatible = "radxa,rockpi4b", "radxa,rockpi4", "rockchip,rk3399";
 };
 
+&pinctrl {
+	usb2 {
+		vcc5v0_host_en: vcc5v0-host-en {
+			rockchip,pins = <4 RK_PD1 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+};
+
 &sdio0 {
 	status = "okay";
 
@@ -43,3 +51,10 @@ bluetooth {
 		vddio-supply = <&vcc_1v8>;
 	};
 };
+
+&vcc5v0_host {
+	enable-active-high;
+	gpio = <&gpio4 RK_PD1 GPIO_ACTIVE_HIGH>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&vcc5v0_host_en>;
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts
index 67344a16821b..5df898013916 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts
@@ -13,6 +13,14 @@ / {
 	compatible = "radxa,rockpi4c", "radxa,rockpi4", "rockchip,rk3399";
 };
 
+&pinctrl {
+	usb2 {
+		vcc5v0_host_en: vcc5v0-host-en {
+			rockchip,pins = <3 RK_PD6 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+};
+
 &sdio0 {
 	status = "okay";
 
@@ -46,9 +54,8 @@ bluetooth {
 };
 
 &vcc5v0_host {
+	enable-active-high;
 	gpio = <&gpio3 RK_PD6 GPIO_ACTIVE_HIGH>;
-};
-
-&vcc5v0_host_en {
-	rockchip,pins = <3 RK_PD6 RK_FUNC_GPIO &pcfg_pull_none>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&vcc5v0_host_en>;
 };
-- 
2.25.1

